x/superfluid/keeper/stake_test.go

```diff
@@ -1028,6 +1032,454 @@ func (s *KeeperTestSuite) TestRefreshIntermediaryDelegationAmounts() {
 	}
 }
 
+func (s *KeeperTestSuite) TestUnbondConvertAndStake() {
+	defaultJoinTime := s.Ctx.BlockTime()
+	type tc struct {
+		notSuperfluidDelegated bool
+		superfluidUndelegating bool
+		unlocking              bool
+		unlocked               bool
+		expectedError          error
+	}
+	testCases := map[string]tc{
+		"lock that is superfluid delegated": {},
+		"lock that is superfluid undelegating": {
+			unlocking:              true,
+			superfluidUndelegating: true,
+		},
+		"bonded lock, not superfluid delegated": {
+			notSuperfluidDelegated: true,
+		},
+		"lock that is unlocking": {
+			unlocking:              true,
+			superfluidUndelegating: true,
+		},
```

Comment:
Same issue here, `lock that is superfluid undelegating` and `lock that is unlocking` have the same input params, so cant be testing different things.

Reply:
My bad, addressed in https://github.com/osmosis-labs/osmosis/pull/5949/commits/0c5b652ee73f06c3fda563502f8a1f8e3b1669b5